### PR TITLE
Add an Action to Setting Index Priority 

### PR DESCRIPTION
**Pull request by bowenlan-amzn**
*Issue #, if available:*
#142 

*Description of changes:*
### possible use case
Sets the priority of the index as soon as the policy enters certain state. Unallocated shards of indices are recovered in order of priority, whenever possible.
[reference from ES](https://www.elastic.co/guide/en/elasticsearch/reference/master/ilm-set-priority.html#ilm-set-priority-options)

```
"actions": [
    {
        "index_priority": {
            "priority": 50
        }
    }
]
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/TestHelpers.kt [hunk 1]
```diff
@@ -24,15 +24,7 @@ import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.Policy
...
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.DeleteActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ForceMergeActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.NotificationActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ReadOnlyActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ReadWriteActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ReplicaCountActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.RolloverActionConfig
-import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.SnapshotActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.*
```
Comment (dbbaughe) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/TestHelpers.kt:27]: np: Don't use * imports

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/indexpriority/AttemptSetIndexPriorityStep.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.step.replicacount
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/indexpriority/AttemptSetIndexPriorityStep.kt:16]: whoops

File: src/main/resources/mappings/opendistro-ism-config.json [hunk 3]
```diff
@@ -175,6 +175,13 @@
                     }
                   }
                 },
+                "set_index_priority": {
```
Comment (dbbaughe) [inline src/main/resources/mappings/opendistro-ism-config.json:178]: change to index_priority

File: src/main/resources/mappings/opendistro-ism-config.json [hunk 4]
```diff
@@ -175,6 +175,13 @@
...
+                  "properties": {
+                    "index_priority": {
```
Comment (dbbaughe) [inline src/main/resources/mappings/opendistro-ism-config.json:180]: can change this to just priority

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/indexpriority/AttemptSetIndexPriorityStep.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            val updateSettingsRequest = UpdateSettingsRequest()
+                    .indices(managedIndexMetaData.index)
+                    .settings(Settings.builder().put("index.priority", indexPriority))
+            val response: AcknowledgedResponse = client.admin().indices()
+                    .suspendUntil { updateSettings(updateSettingsRequest, it) }
+
+            if (response.isAcknowledged) {
+                logger.info("Successfully executed $name on ${managedIndexMetaData.index}")
+                stepStatus = StepStatus.COMPLETED
+                info = mapOf("message" to "Set index_priority to $indexPriority")
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/indexpriority/AttemptSetIndexPriorityStep.kt:57]: Can we change message to something like "Successfully set ..."

File: src/main/resources/mappings/opendistro-ism-config.json (hunk 3 above)
Comment (bowenlan-amzn) [reply src/main/resources/mappings/opendistro-ism-config.json:178]: 😅

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/indexpriority/AttemptSetIndexPriorityStep.kt (hunk 2 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/indexpriority/AttemptSetIndexPriorityStep.kt:16]: 😅

File: src/main/resources/mappings/opendistro-ism-config.json [hunk 6]
```diff
@@ -175,6 +175,13 @@
                     }
                   }
                 },
+                "index_priority": {
```
Comment (dbbaughe) [inline src/main/resources/mappings/opendistro-ism-config.json:178]: There is also a "schema_version" in this file at the top that you need to bump

**Comment by bowenlan-amzn** [comment]
already checked this has no conflict with opendistro-1.9 update, safe to merge in

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/AttemptSetIndexPriorityStepTests.kt [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            val managedIndexMetaData = ManagedIndexMetaData("test", "indexUuid", "policy_id", null, null, null, null, null, null, null, null, null, null)
+            val attemptSetPriorityStep = AttemptSetIndexPriorityStep(clusterService, client, indexPriorityActionConfig, managedIndexMetaData)
+            attemptSetPriorityStep.execute()
+            val updatedManagedIndexMetaData = attemptSetPriorityStep.getUpdatedManagedIndexMetaData(managedIndexMetaData)
+            logger.info(updatedManagedIndexMetaData)
+            assertEquals("Step status is not FAILED", Step.StepStatus.FAILED, updatedManagedIndexMetaData.stepMetaData?.stepStatus)
+        }
+    }
+
+    private fun getClient(adminClient: AdminClient): Client = mock { on { admin() } doReturn adminClient }
```
Comment (qreshi) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/AttemptSetIndexPriorityStepTests.kt:83]: NP: Missing newlines in between function definitions

**Pull request merged**